packages/client-common/src/result.ts

```diff
@@ -1,4 +1,40 @@
-export interface Row {
+import type { ResponseJSON } from './clickhouse_types'
+import type {
+  DataFormat,
+  RawDataFormat,
+  RecordsJSONFormat,
+  SingleDocumentJSONFormat,
+  StreamableJSONDataFormat,
+} from './data_formatter'
+
+export type ResultJSONType<T, F extends DataFormat | unknown> =
```

Comment:
Maybe it is too much, but we might need some tests for such complex types. Or do you think the current JS tests in TypeScript cover all the cases?  https://effectivetypescript.com/2022/05/28/eslint-plugin-expect-type/

Reply:
I think this is a good idea, it's not too much to test - basically, a few calls for every DataFormat, and that's it. Could be put into some unit test that expects nothing, and just serves as an ESLint check.